src/Transformer.ts

```diff
@@ -0,0 +1,400 @@
+import * as path from "path";
+import * as ts from "typescript";
+
+import {CompilerOptions} from "./CompilerOptions";
+import {DecoratorKind} from "./Decorator";
+import {TSTLErrors} from "./Errors";
+import {TransformHelper as transformHelper} from "./TransformHelper";
+import {TSHelper as tsHelper} from "./TSHelper";
+
+export class LuaTransformer {
+    private checker: ts.TypeChecker;
+    private options: CompilerOptions;
+    private context: ts.TransformationContext;
+
+    constructor(checker: ts.TypeChecker, options: CompilerOptions) {
+        this.checker = checker;
+        this.options = options;
+    }
+
+    public transform(node: ts.SourceFile): ts.SourceFile {
+        return ts
+            .transform(node,
+                       [(ctx: ts.TransformationContext) => {
+                           this.context = ctx;
```

Comment:
Agreed, we should replicate the pattern we use for transpiler targets (TranspilerFactory, createTranspiler...). Since we most likely want to have differentiations transformers for differentiations Lua versions in the future. But I would reorganise that some point later once we are sure how exactly the transformer architecture will be implemented.